### PR TITLE
[Method References 2] LPS-167249 Convert method reference to ServiceTrackerList in module portal-search-web

### PR DESCRIPTION
**Pull request by W2jx**

### CODE REVIEW

**Comment by W2jx** [comment]
ci:test:search

File: modules/apps/portal-search/portal-search-web/src/main/java/com/liferay/portal/search/web/internal/search/results/portlet/SearchResultsPortlet.java [hunk 1]
```diff
@@ -128,14 +129,15 @@ public void render(
...
-	@Reference(
-		cardinality = ReferenceCardinality.MULTIPLE,
-		policy = ReferencePolicy.DYNAMIC
-	)
-	protected void addSearchResultImageContributor(
-		SearchResultImageContributor searchResultImageContributor) {
+	@Activate
+	protected void activate(BundleContext bundleContext) {
+		_serviceTrackerList = ServiceTrackerListFactory.open(
+			bundleContext, SearchResultImageContributor.class);
```
Comment (dantewang) [inline modules/apps/portal-search/portal-search-web/src/main/java/com/liferay/portal/search/web/internal/search/results/portlet/SearchResultsPortlet.java:135]: @W2jx This interface doesn't have any implementation. Could you ask the Search team about this?

**Comment by catarineaguiar** [comment]
@W2jx regressions found:

`SearchDisplayContextTest`, `SearchResultsPortletTest`, ``
```
 [exec] com.liferay.portal.search.web.internal.display.context.SearchDisplayContextTest > testNoScopeParameter FAILED
     [exec]     java.lang.NullPointerException
     [exec]         at com.liferay.portal.search.web.internal.facet.SearchFacetRegistry.getSearchFacets(SearchFacetRegistry.java:35)
     [exec]         at com.liferay.portal.search.web.internal.display.context.SearchDisplayContext.getSearchFacets(SearchDisplayContext.java:338)
     [exec]         at com.liferay.portal.search.web.internal.display.context.SearchDisplayContext.getEnabledSearchFacets(SearchDisplayContext.java:210)
     [exec]         at com.liferay.portal.search.web.internal.display.context.SearchDisplayContext._addEnabledSearchFacets(SearchDisplayContext.java:555)
     [exec]         at com.liferay.portal.search.web.internal.display.context.SearchDisplayContext._contributeSearchSettings(SearchDisplayContext.java:587)
     [exec]         at com.liferay.portal.search.web.internal.search.request.SearchRequestImpl.lambda$_buildSettings$1(SearchRequestImpl.java:121)
```
full log [here](https://testray.liferay.com/reports/production/logs/2022-11/test-1-38/test-portal-acceptance-pullrequest(master)/4995/modules-unit-jdk8/0/0/jenkins-console.txt.gz?authuser=0)

File: modules/apps/portal-search/portal-search-web/src/main/java/com/liferay/portal/search/web/internal/search/results/portlet/SearchResultsPortlet.java (hunk 1 above)
Comment (W2jx) [reply modules/apps/portal-search/portal-search-web/src/main/java/com/liferay/portal/search/web/internal/search/results/portlet/SearchResultsPortlet.java:135]: Hi @dantewang , I'll ask them.

**Pull request closed without merging**